include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -166,6 +166,72 @@ def XTenNN_DequantizeOp: XTenNN_Op<"dequantize", [
   let assemblyFormat = [{ `(`$input `:` type($input)`)` attr-dict `->` type($output) }];
 }
 
+def XTenNN_GroupQuantizeOp: XTenNN_Op<"group_quantize", [
+            Pure]> {
+  let summary = "Group quantizes a float32 tensor to an integer tensor of given width.";
+  let description = [{
+    Quantizes a given float32 tensor into a integer tensor of given width.
+
+    Scales and zeros must have a shape that is broadcastable to the input shape.
+    Input and scales most have the same dtype, as must zeros and quants.
+    The range of values in zeros and quants is given by the min and max attributes,
```

Comment:
zeros and quants does that refer to the operand name or the formula below?

Also the min, max and bits attributes is that different to the output type?

Reply:
`zeros`and `quants` are the names in `arguments` and `results` (and the formula).

`min` and `max`are copied from the `Clip` and can be a smaller range that `bits` would allow. The output type may be wider than `bits` because I've hit some complications getting the current conversion process to work without this. I hope we can improve, but for now I'd like to live with this.